src/datachain/lib/dc/storage.py

```diff
@@ -10,15 +10,21 @@
     FileType,
     get_file_type,
 )
-from datachain.lib.listing import get_file_info, get_listing, list_bucket, ls
+from datachain.lib.listing import (
+    get_file_info,
+    get_listing,
+    list_bucket,
+    ls,
+    parse_listing_uri,
+)
 from datachain.query import Session
 
 if TYPE_CHECKING:
     from .datachain import DataChain
 
 
 def from_storage(
-    uri: Union[str, os.PathLike[str]],
+    uri: Union[str, os.PathLike[str], list[str], list[os.PathLike[str]]],
     *,
```

Comment:
Should we use `Iterable` or `Sequence` instead of `list` type to also allow `tuple`, for example?

Also just wonder if we can use `*args` for this.

```python
dc.from_storage("s3://bucket1/dir1")
```

and

```python
dc.from_storage("s3://bucket1/dir1", "s3://bucket1/dir2")
```

instead of using lists:

```python
dc.from_storage(["s3://bucket1/dir1", "s3://bucket1/dir2"])
```